src/TaggableCachePoolTest.php

```diff
@@ -292,6 +275,20 @@ public function testInvalidateTags()
         $this->cache->save($item);
         $this->cache->invalidateTags(['tag1']);
 
-        $this->assertTrue($this->cache->hasItem('key'), 'Item key list should be removed when clearing the tags');
+        $this->assertTrue($this->cache->hasItem('key'), 'Item k list should be removed when clearing the tags');
+    }
+
+    /**
+     * When an item is overwritten we need to clear tags for original item.
+     */
+    public function testTagsAreCleanedOnSave()
+    {
+        $pool = $this->cache;
+        $i    = $pool->getItem('key');
+        $pool->save($i->addTag('foo'));
+        $i = $pool->getItem('key');
+        $pool->save($i->addTag('bar'));
+        $pool->invalidateTags(['foo']);
+        $this->assertTrue($pool->getItem('key')->isHit());
```

Comment:
Is this test correct? Or... is this the expected behavior? If I fetch an object from cache and then _add_ some tags to it. It should keep the existing tags, right?

The [Symfony test](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Cache/Tests/Adapter/TagAwareAdapterTest.php#L81) is using the `tag` function which has different features. 

I would say that this is something the tag PSR should specify. I want to change this test to: `$this->assertFalse($pool->getItem('key')->isHit());`